src/aind_hcr_data_transformation/compress/czi_to_zarr_s3.py

```
@@ -379,7 +382,15 @@ def czi_stack_zarr_writer(
                 slice(0, dataset_shape[-2]),
                 slice(0, dataset_shape[-1]),
             )
-            dataset[region].write(pad_array_n_d(block)).result()
+            try:
+                with ts.Transaction() as transaction: 
+                    dataset[region].with_transaction(transaction).write(pad_array_n_d(block)).result()
+                block_count += 1
+                logging.info(f"Completed block {block_count} write for z-slices {axis_area}")
+            except Exception as e:
```

I think we should just let the job fail.